### ISSUE TITLE
npm install fails with 404 error not found in registry

### ISSUE THREAD

**Issue by ssfica** [s1hofmann/native-ui-toolkit#592]
**Version**

**On a VM on which Jenkins is configured to run automated tests from playwright automated solution it started to return 404 since 30th of April 2024 whenever it's trying to do an npm install**

**Issue occurs on**

- [ ] Virtual machine
- [ ] @nut-tree/nut-js@3.1.2' is not in this registry
![image](https://github.com/nut-tree/nut.js/assets/100785000/f99eac31-7002-4104-b23a-983bbd08400b)

**node version**:

**OS type and version**:

**Detailed error description**

C:\jenkins\workspace\cc-oe-tests-playwright\cc-oe-sanity-tests-playwright\e2e-tests>npm install 
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@nut-tree%2fnut-js - Not found
npm ERR! 404 
npm ERR! 404  '@nut-tree/nut-js@3.1.2' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

**Full code sample to reproduce**
In order to see this error the steps executed in jenkins are:
clone repo
npm install
command for test run

**Additional content**
https://registry.npmjs.org/@nut-tree%2fnut-js

> Please provide any (mandatory) additional data to reproduce the error (Dockerfiles etc.)

**Comment by ssfica** [comment]
I've tried it on the VM after a cleanup, somehow the test run on the VM simulates a new setup each time the test run is triggered and I could reproduce the error manually.
Not sure if there is any workaround fro the 'npm install step'?

**Comment by s1hofmann** [comment]
Hi @ssfica 👋 

a series of events has lead to nut.js no longer being publicly available on npm.

Pre-built packages are now only available via the nut.js package registry. Please read [this blog post](https://nutjs.dev/blog/i-give-up) to learn about the reasons for this.

You’ll have to build and package nut.js yourself from sources, or, if you want to have continuous updates and ready-to-use packages, choose a subscription [here](https://nutjs.dev/pricing/pricing).

Best regards 

Simon

**Comment by ssfica** [comment]
HI @s1hofmann , thank you for your reply!
Best regards, 
Simina

**Comment by arift** [comment]
Do you have any plans to bring back existing packages that were deployed to npm? I understand and respect your decision to no longer release easily consumable npm packages and I appreciate the fact that you left the repo public, but pulling existing versions from the registry is intentionally being hurtful to those who was dependent on it. It seems a bit harsh, but it is your project.

**Comment by s1hofmann** [comment]
Hi @arift 👋 

I would certainly do that if we could establish a way to make it work for both sides.

Providing something for free only works as long as it's affordable, and I realised that I'm no longer in that position.
Free packages have been an open invitation for very entitled people to behave like absolute morons, and I've reached a point where I'm absolutely no longer willing to put up with this, because my own time is far to valuable.

The current way of "build it yourself from sources, for free" vs. "pay me for ready made packages" is an approach that I still consider fair, because you don't have to spend anything in exchange for your own time to build an package the project. On the other hand, these paid packages allow me to buy back time in real life to keep working on nut.js, but drastically reduce the amount of complaints I have to deal with. It's essentially nothing else but a sponsoring tier with an attached benefit.

So to sum things up:

Yes, I'd open up packages again if

- I would no longer have to deal with insults an blatant accusations
- the community, and especially companies that use nut.js commercially, would support this project in a way that makes it possible to work on it without sacrificing my sleep schedule

If that's not the case I'd rather keep things the way they are now, because I prefer to work with people that use nut.js because it brings them value, not because it's free.

**Comment by arift** [comment]
Thanks for the detailed response. It's unfortunate that the few entitled people on the internet ruined it for everyone else. 
In any case, thanks again for your effort and leaving the project as public despite the difficulties it brought.
